### PR TITLE
add config option `build_as_root`

### PR DESCRIPTION
**Pull request by pkit**
new config option that does not drop to unprivileged user at build time
helps to build some legacy packages that expect to run build tests as root

### CODE REVIEW

File: py/mockbuild/buildroot.py [hunk 1]
```diff
@@ -175,17 +175,17 @@ def doChroot(self, command, shell=False, nosync=False, *args, **kargs):
         env = dict(self.env)
         if nosync and self.nosync_path:
             env['LD_PRELOAD'] = self.nosync_path
-        if util.USE_NSPAWN:
+        if util.USE_NSPAWN or kargs.get('uid', 1000) == 0:
```
Comment (xsuchy) [inline py/mockbuild/buildroot.py]: Why is this here?

Comment (xsuchy) [inline py/mockbuild/buildroot.py]: I meant - why are you testing for uid == 1000? The uid of user can be different. E.g. 1005. I understand the change in backend.py. But this one I cannot comprehend.

**Comment by xsuchy** [comment]
Can you add this option to etc/mock/site-config.cfg (as comment) - with comment what it does? And add code to utils.py which will set it to defaults.

File: py/mockbuild/buildroot.py (hunk 1 above)
Comment (pkit) [inline py/mockbuild/buildroot.py]: It's just a placeholder. I could probably use `-1` here as well.
If we elevated to root `kargs['uid'] == 0` otherwise kargs may not even have 'uid' key (yes, it does happen).
But if `-1` is better, I can change it.

**Comment by pkit** [comment]
Fixed the following:

- placeholder uid
- utils
- config example and comments

**Comment by xsuchy** [comment]
This has been obsoleted by https://github.com/rpm-software-management/mock/pull/193 which allows you to build as root now. Thank you for this PR anyway.
